pkg/idtools/idtools_unix.go

```diff
@@ -27,20 +27,19 @@ func mkdirAs(path string, mode os.FileMode, ownerUID, ownerGID int, mkAll, chown
 	// chown the full directory path if it exists
 	var paths []string
 
-	stat, err := system.Stat(path)
-	if err == nil {
+	if stat, err := system.Stat(path); err != nil {
+		if !os.IsNotExist(err) {
+			return err
```

Comment:
This is new; previously any error that was _not_ an `os.IsNotExist()` was ignored; I think this is correct, but double check if there was a reason we ignored those errors

Reply:
I would guess it is because the `Stat` is not really important. It's just a check to see if the operation can be skipped, a failure is acceptable here. If the path is invalid then the `mkdir()` below will error out with a more appropriate error message.

Reply:
Ah, yes, you may be right; returning this error won't add much. Let me revert this part